src/zope/app/apidoc/typemodule/type.py

```diff
@@ -69,10 +69,11 @@ def items(self):
         """See zope.container.interfaces.IReadContainer"""
         results = [(name, LocationProxy(iface, self, name))
                    for name, iface in getUtilitiesFor(self.interface)]
-        results.sort(lambda x, y: cmp(x[1].getName(), y[1].getName()))
+        results.sort(key=lambda x: x[1].__name__)
```

Comment:
This makes no sense to me. `x[1].__name__` should always be same as `x[0]`, if LocationProxy still works the way I remember it working.

AFAIU the old code was sorting by the name of the interface.  The new code is sorting by the name of the utility.  Many utilities are unnamed (which means their name is either `""` or `None`; I don't remember which).  Interfaces always have names.  IIRC the values of `iface` here can be subinterfaces of `self.interface`, so they're not necessarily the same.

(Also, I don't see why we're sorting by just one of the two names.  It would make sense to me to sort on both.)